allensdk/brain_observatory/behavior/session_apis/data_transforms/behavior_data_xforms.py

```diff
@@ -18,130 +19,11 @@
-        :rtype: dict
-        """
-        # Get all ids from the behavior_sessions table
-        query = f"""
-            SELECT
-                ophys_session_id, behavior_training_id, foraging_id
-            FROM
-                behavior_sessions
-            WHERE
-                behavior_sessions.id = {self.behavior_session_id};
-        """
-        ids_response = self.lims_db.select(query)
-        if len(ids_response) > 1:
-            raise OneResultExpectedError
-        ids_dict = ids_response.iloc[0].to_dict()
-
-        #  Get additional ids if also an ophys session
-        #     (experiment_id, container_id)
-        if ids_dict.get("ophys_session_id"):
-            oed_query = f"""
-                SELECT id
-                FROM ophys_experiments
-                WHERE ophys_session_id = {ids_dict["ophys_session_id"]};
-                """
-            oed = self.lims_db.fetchall(oed_query)
-            if len(oed) == 0:
-                oed = None
-
-            container_query = f"""
-            SELECT DISTINCT
-                visual_behavior_experiment_container_id id
-            FROM
-                ophys_experiments_visual_behavior_experiment_containers
-            WHERE
-                ophys_experiment_id IN ({",".join(set(map(str, oed)))});
-            """
-            try:
-                container_id = self.lims_db.fetchone(container_query, strict=True)
-            except OneResultExpectedError:
-                container_id = None
-
-            ids_dict.update({"ophys_experiment_ids": oed,
-                             "ophys_container_id": container_id})
-        else:
-            ids_dict.update({"ophys_experiment_ids": None,
-                             "ophys_container_id": None})
-        return ids_dict
+from allensdk.brain_observatory.behavior.trials_processing import (
+    get_trials, get_extended_trials)
 
-    def get_behavior_session_id(self) -> int:
-        """Getter to be consistent with BehaviorOphysLimsApi."""
-        return self.behavior_session_id
 
-    def get_behavior_session_uuid(self) -> Optional[int]:
-        data = self._behavior_stimulus_file()
-        return data.get("session_uuid")
-
-    def get_behavior_stimulus_file(self) -> str:
-        """Return the path to the StimulusPickle file for a session.
-        :rtype: str
-        """
-        query = f"""
-            SELECT
-                stim.storage_directory || stim.filename AS stim_file
-            FROM
-                well_known_files stim
-            WHERE
-                stim.attachable_id = {self.behavior_session_id}
-                AND stim.attachable_type = 'BehaviorSession'
-                AND stim.well_known_file_type_id IN (
-                    SELECT id
-                    FROM well_known_file_types
-                    WHERE name = 'StimulusPickle');
-        """
-        return safe_system_path(self.lims_db.fetchone(query, strict=True))
+class BehaviorDataXforms(BehaviorBase):
```

Comment:
Why `Xforms` and not `Transforms`? Also, what about this is transformative at all? It is almost all `get()`?

docstring for class?

Reply:
Yeah, I agree with Dan that the separation here isn't totally clear to me. This class is still very dependent on the LIMS data (the using the pkl file) and won't generalize to NWB.

Reply:
I wanted to separate methods that get data from LIMS vs methods that operate on the data obtained from LIMS.

These methods are not supposed to generalize to NWB as the flow of data looks like:

1. Fill a BehaviorSession or BehaviorOphysSession with a "data + transform/xform" API from either a LIMS api or from a json api.

2. Once you have a BehaviorSession/BehaviorOphysSession, save data into NWB format.

3. Once you have an NWB format file, you can load directly from that NWB and get a functional BehaviorSession/BehaviorOphysSession without needing to interact with LIMS/JSON APIs anymore.

Reply:
ok